index.js

```diff
@@ -17,3 +17,103 @@ for (let menu of menuitems) {
+          <div class="">
+            <div class="canopy">
+              <span></span>
+              <ul class="canopy-list">
+                <li class="backend"></li>
+                <li class="year"></li>
+              </ul>
+            </div>
+
+           <div class="img">
+              <img src= alt="" />
+            </div>
+          <div class="text-container">
+
+              <div class="parag"><p class="p1"></p></div>
+             <div class="buttons">
+              <ul class="list-B">
+                <li class="">HTML</li>
+                <li class="">Css</li>
+                <li class="">JavaScript</li>
+                <li class="ruby">Ruby</li>
+                <li class="bootstrap">Boostrap</li>
+              </ul>
+            
+              <button  class="btn1 btn-1">See Live<img src="./images/icon.png" alt="" class="icon-img"></button>
+              <button class="btn2 btn-1">See source<i class="fa-brands fa-github"></i></button>
+              
+              </div>
+            </div>
+
+          </div>
+            
+        </div>
+      </div>`;
+
+div.className = "popupcontainer";
+
+popupwindow.append(div);
+
+const popclose = document.querySelector(".icon-close");
+popclose.addEventListener("click", function () {
+  popupwindow.classList.remove("visible");
+});
+
+const projectbtn = document.querySelectorAll(".btn");
+for (let btn of projectbtn) {
+  btn.addEventListener("click", function (e) {
+    e.stopPropagation();
+    popupwindow.classList.add("visible");
+
+    const array = [];
+
+    const card = e.target.parentNode.parentNode.parentNode;
+
+    const h1 = card.querySelector(".h-1").innerHTML;
+    const span = card.querySelector(".project span").innerHTML;
+    const backend = card.querySelectorAll(".listA-items");
+    const img = card.querySelector(".cards img").src;
+    const paragraph = card.querySelector(".p-1");
+    console.log(paragraph);
+
+    const portfolio = card.querySelectorAll(".listB-items");
+
+    array[0] = h1;
+    array[1] = span;
+    array[2] = { list1: backend[0].innerHTML, list2: backend[1].innerHTML };
+    array[3] = img;
+    array[4] = `Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s. Lorem Ipsum is simply dummy text of the printing and typesetting industry. 
+
+Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s with the releorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum han printer took a galley of type and scrambled it 1960s with the releawn printer took a galley of type and scrambled it 1960s.`;
+    array[5] = {
+      portfolio1: portfolio[0].innerHTML,
+      portfolio2: portfolio[1].innerHTML,
+      portfolio3: portfolio[2].innerHTML,
+    };
```

Comment:
- [ ] Good job  fetching the data from DOM 👏 , but you should have these data stored in an array 👍 , with predefined data like: name, description, featured image, technologies, link to live version, link to the source
I really appreciate your outlook on this point but fo good practice Kindly improve 👍 

```js
const projects = [{
    name: "Submarine game",
    description: "A Phaser3 platform game built as part of the Microverse JavaScript program.",
    image: "link_to_image.jpg",
    technologies: ["Phaser3", "JavaScript", "Jest"],
    link: "link_to_live_version.com",
}]
```

Reply:
Thank you for your comments. Just to mention that except you wish that I change the names of the variables to match as mentioned in the project description otherwise if you look carefully, you will notice that, after fetching the data from the DOM, these are actually stored in an array called "array" set to an empty array in line 78. It's from these arrays that the data if finally sent to the popup window. Please reply for appropriate action. Cheers